components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/ToolbarInteractor.kt

```diff
@@ -0,0 +1,34 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.toolbar
+
+import mozilla.components.concept.toolbar.Toolbar
+import mozilla.components.feature.session.SessionUseCases
+import mozilla.components.support.ktx.kotlin.toNormalizedUrl
+
+/**
+ * Connects a toolbar instance to the browser engine via use cases
+ */
+class ToolbarInteractor (
+    private val loadUrlUseCase: SessionUseCases.LoadUrlUseCase,
+    private val toolbar: Toolbar
+) {
+
+    /**
+     * Starts this interactor. Makes sure this interactor is listening
+     * to relevant UI changes and triggers the corresponding use-cases
+     * in response.
+     */
+    fun start() {
+        toolbar.setOnUrlChangeListener { url -> loadUrlUseCase.invoke(url.toNormalizedUrl()) }
+    }
+
+    /**
+     * Stops this interactor.
+     */
+    fun stop() {
+        toolbar.setOnUrlChangeListener { }
```

Comment:
@pocmo I wasn't sure but ran a decompiler now. 

Basically, I wanted to use an empty lambda to prevent setting the listener to null and making it a nullable type. 

Here's how the start method gets translated into its own class: https://gist.github.com/csadilek/0a2afd8a7ed4e203060a486fca032414 (note how it references the ToolbarInteractor)

And here's the stop method (doesn't reference the ToolbarInteractor):
https://gist.github.com/csadilek/de157d77eddb423e8f21c784ce27cd76

So, this looks like we're fine.